src/main/java/cms/gov/madie/measure/config/mongock/AddVersionDateAndCreatedBeforeVersioningChangeUnit.java

```diff
@@ -0,0 +1,50 @@
+package cms.gov.madie.measure.config.mongock;
+
+import cms.gov.madie.measure.repositories.MeasureRepository;
+import gov.cms.madie.models.measure.Measure;
+import gov.cms.madie.models.measure.TestCase;
+import io.mongock.api.annotations.ChangeUnit;
+import io.mongock.api.annotations.Execution;
+import io.mongock.api.annotations.RollbackExecution;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+
+import java.util.List;
+
+@Slf4j
+@ChangeUnit(
+    id = "add_version_date_and_created_before_versioning",
+    order = "1",
+    author = "madie_dev")
+public class AddVersionDateAndCreatedBeforeVersioningChangeUnit {
+
+  @Execution
+  public void addVersionDateAndCreatedBeforeVersioningChangeUnit(
+      MeasureRepository measureRepository) {
+    List<Measure> measures = measureRepository.findAll();
+    measures.forEach(
+        measure -> {
+          if (!measure.getMeasureMetaData().isDraft()) {
+            measure.getMeasureMetaData().setVersionDate(measure.getLastModifiedAt());
+
+            List<TestCase> testCases = measure.getTestCases();
+            if (CollectionUtils.isEmpty(testCases)) {
+              return;
+            }
+
+            testCases.stream()
+                .forEach(
+                    testCase -> {
+                      testCase.setCreatedBeforeVersioning(true);
+                    });
+            measureRepository.save(measure);
```

Comment:
Even if there are no test cases for versioned measure which is highly unlikely, I think we would still save the measure with versionDate
```suggestion
            if (!CollectionUtils.isEmpty(testCases)) {
              testCases.stream()
                .forEach(
                    testCase -> {
                      testCase.setCreatedBeforeVersioning(true);
                    });
            }
            measureRepository.save(measure);
```

Reply:
you're right, that makes sense. thanks for the feedback Ashok, I just pushed the code with the changes